### ISSUE TITLE
helm-comint problem with Emacs < 29.1

### ISSUE THREAD

**Issue by spt29** [syl20bnr/spacemacs#16169]
The commit 195e323693ba229e82586dc77adf51f3ebb64a1b introduced an error at startup with Emacs 28.1. From the `*Messages*` buffer
```
...
Contacting host: elpa.nongnu.org:443
Package refresh done
(Spacemacs) --> installing package: helm-comint@helm... [1/1]
(Spacemacs) Error: 
An error occurred while installing helm-comint (error: (error Package ‘emacs-29.1’ is unavailable))
```
This happens at _every_ startup now, commenting out `helm-comint` from `helm-packages` (see the commit above) makes things work again. It looks like `helm-comint` is unusable for non-bleeding-edge Emacs versions. I am not sure what that fix was trying to fix, things seem to work for me without it.

**Comment by gabubarks** [comment]
+1, also on 28.1 and have the same problem after updating today

**Comment by pataquets** [comment]
Same here, right after updating.
Also, leaving the question as novice Elisp'er:
What would be the _inline_ syntax for the list element `helm-comint` [here](https://github.com/syl20bnr/spacemacs/commit/195e323693ba229e82586dc77adf51f3ebb64a1b#diff-227a4156f81107ef073c3bfca9b84e42ce225bc8d50c27a2bf53efeefe132318R38) to be conditionally included only if Emacs >= 29?

**Comment by smile13241324** [comment]
Hmmm, I have actually just fixed a PR I have merged yesterday which was not properly loading helm-comint in shell layer.

I was not sure that this is breaking emacs 28 which is undortunate, still please remember that emacs 29 is definitely not bleeding edge and fixes a lot of issues in package.el which are simply necessary.

With this being said a simple when as if condition in the package declaration should fix the issue. Search for emacs-major-version.

I will provide a fix when I am back at home.

**Comment by spt29** [comment]
Well, I definitely consider 29.1 "bleeding edge". If you look e.g. at Ubuntu releases, the current LTS still ships with 27.1, the non-LTS after that with 28.2, and only the very latest non-LTS (only 2 weeks old) ships with 29.1. Not everybody is compiling emacs from source or is free to choose the distro at work, so I would very much appreciate keeping things compatible with some older Emacs versions.

Thanks for looking into this! :smiley:

**Comment by jaeyeom** [comment]
This bug helped me find that one of my systems Debian Bookworm was using Emacs 28. I've upgraded to Emacs 29.1 using backport package.

**Comment by smile13241324** [comment]
Should be fixed please reopen if not

**Comment by pataquets** [comment]
Confirming it's fixed (Emacs 28.1). Thank you @smile13241324.

**Comment by sunlin7** [comment]
Please help verify the better solution: https://github.com/syl20bnr/spacemacs/pull/16177.
Thanks.
